gateway/crates/rolling-logger/src/strategy.rs

```diff
@@ -0,0 +1,78 @@
+use std::time::{Duration, SystemTime};
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) enum StrategyKind {
+    Never,
+    Minutely,
+    Hourly,
+    Daily,
+    MaxSize(u64),
```

Comment:
Nitpick, but with the max size strategy, the log files are going to grow slightly larger than max size in most cases, since it rotates only after max size is exceeded. Maybe rename to size based?